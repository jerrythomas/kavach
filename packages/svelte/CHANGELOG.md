# @kavach/svelte

## 1.0.0-next.6

### Patch Changes

- Added store for auth status and used alerts to display status on demo page.
- Updated dependencies
  - @kavach/cookie@1.0.0-next.6
  - @kavach/core@1.0.0-next.6
  - @kavach/deflector@1.0.0-next.6
  - @kavach/logger@1.0.0-next.6

## 1.0.0-next.5

### Patch Changes

- chore: release 1.0.0-next.4
- Split core package into logger and deflector
- 1a35ad0: Resolved linter errors.
- Releasing the next version.
- Updated dependencies
- Updated dependencies
- Updated dependencies [1a35ad0]
- Updated dependencies
  - @kavach/cookie@1.0.0-next.5
  - @kavach/core@1.0.0-next.5
  - @kavach/deflector@1.0.0-next.5
  - @kavach/logger@1.0.0-next.5
