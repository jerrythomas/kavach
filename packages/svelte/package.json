{
	"name": "@kavach/svelte",
	"version": "1.0.0-next.2",
	"description": "Drop in security for web apps.",
	"author": "Jerry Thomas <me@jerrythomas.name>",
	"license": "MIT",
	"main": "index.js",
	"svelte": "src/index.js",
	"module": "src/index.js",
	"types": "dist/index.d.ts",
	"type": "module",
	"repository": {
		"type": "git",
		"url": "https://github.com/jerrythomas/kavach",
		"directory": "packages/kavach"
	},
	"prettier": "@jerrythomas/prettier-config",
	"publishConfig": {
		"access": "public"
	},
	"scripts": {
		"prepublishOnly": "tsc",
		"lint": "prettier --check --plugin-search-dir=. . && eslint .",
		"format": "prettier --write --plugin-search-dir=. .",
		"test:ci": "vitest run",
		"test:ui": "vitest --ui",
		"test": "vitest",
		"coverage": "vitest run --coverage",
		"upgrade": "pnpm upgrade"
	},
	"devDependencies": {
		"@jerrythomas/prettier-config": "^1.0.0",
		"@sveltejs/vite-plugin-svelte": "^2.0.2",
		"@testing-library/svelte": "^3.2.2",
		"@vitest/ui": "~0.12.10",
		"c8": "^7.12.0",
		"eslint-config-svelte": "workspace:*",
		"js-yaml": "^4.1.0",
		"jsdom": "^19.0.0",
		"shared-config": "workspace:^1.0.0",
		"svelte": "^3.55.0",
		"typescript": "^4.9.4",
		"vite": "^4.0.3",
		"vitest": "~0.19.1"
	},
	"exports": {
		"./package.json": "./package.json",
		".": {
			"types": "./types/index.d.ts",
			"import": "./src/index.js"
		}
	},
	"dependencies": {
		"@kavach/cookie": "workspace:1.0.0-next.2",
		"@kavach/core": "workspace:*",
		"@svelte-spice/form": "1.0.0-next.3",
		"@svelte-spice/input": "1.0.0-next.3",
		"ramda": "^0.28.0"
	}
}
