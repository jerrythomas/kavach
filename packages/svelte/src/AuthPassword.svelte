<script>
	import { InputField } from '@rokkit/organisms'
	import { createEventDispatcher } from '@svelte'

	const dispatch = createEventDispatcher()

	export let type = 'email'
	export let value = ''
	export let password = ''

	function handle() {
		dispatch('click', { [type]: value, password })
	}
</script>

<InputField {type} icon="logo-{type}" label={type} bind:value />
<InputField type="password" icon="logo-{type}" label="Password" bind:password />
<button
	on:click={handle}
	class="col-start-3 text-center h-10 mt-4 bg-primary text-white"
>
	Sign in
</button>
