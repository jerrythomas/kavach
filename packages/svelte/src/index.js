export * from './kavach'
export * from './request'
export * from './endpoints'

export { default as AuthButton } from './AuthButton.svelte'
export { default as AuthProvider } from './AuthProvider.svelte'
export { default as AuthGroup } from './AuthGroup.svelte'
export { default as AuthError } from './AuthError.svelte'
export { default as AuthResponse } from './AuthResponse.svelte'
export { default as AuthHandler } from './AuthHandler.svelte'
export { default as Logout } from './Logout.svelte'
