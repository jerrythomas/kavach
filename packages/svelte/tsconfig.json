{
  "extends": "shared-config/jsconfig.json",
  "include": ["src/**/*.js", "src/**/*.svelte"],
  "compilerOptions": {
    "outDir": "dist"
  }
}
