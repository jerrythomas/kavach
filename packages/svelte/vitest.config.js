import { defineConfig } from 'vite'
import { svelte } from '@sveltejs/vite-plugin-svelte'
import { testConfig } from 'shared-config'

testConfig.coverage.provider = 'v8'
export default defineConfig({
	plugins: [svelte({ hot: !process.env.VITEST })],
	test: testConfig
})
