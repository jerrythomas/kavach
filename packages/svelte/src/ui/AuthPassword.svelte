<script>
	import { InputField } from '@rokkit/organisms'
	import { createEventDispatcher } from 'svelte'

	const dispatch = createEventDispatcher()

	export let type = 'email'
	export let value = ''
	export let password = ''

	function handle() {
		dispatch('click', { [type]: value, password })
	}

	$: type = type === 'phone' ? 'tel' : type
</script>

<InputField
	name="email"
	{type}
	icon="i-auth-{type}"
	label={type}
	autocomplete="on"
	bind:value
/>
<InputField
	name="password"
	type="password"
	icon="i-auth-password"
	label="Password"
	bind:password
/>
<button
	on:click={handle}
	class="col-start-3 text-center h-10 mt-4 bg-primary text-white"
>
	Sign in
</button>
