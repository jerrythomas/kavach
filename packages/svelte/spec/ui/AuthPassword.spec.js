import { describe, expect, it, beforeEach, vi } from 'vitest'
import { cleanup, fireEvent, render } from '@testing-library/svelte'
import { userEvent } from '@testing-library/user-event'
import { tick } from 'svelte'
import { toHaveBeenDispatchedWith } from './validators'
import AuthPassword from '../../src/ui/AuthPassword.svelte'

expect.extend({ toHaveBeenDispatchedWith })
describe('AuthPassword.svelte', () => {
	const handle = vi.fn()

	beforeEach(() => {
		// userEvent.setup()
		cleanup()
		handle.mockClear()
	})

	it('should render email/password auth', async () => {
		const { container, component } = render(AuthPassword)
		expect(container).toMatchSnapshot()

		component.$on('click', handle)
		const emailInput = container.querySelector('input[type="email"]')
		const passwordInput = container.querySelector('input#password')
		const button = container.querySelector('button')

		await userEvent.type(emailInput, 'john.doe@example.com')
		await userEvent.type(passwordInput, 'password') // password value does not change

		await fireEvent.click(button)
		await tick()
		expect(handle).toHaveBeenDispatchedWith({
			email: 'john.doe@example.com',
			password: ''
		})
	})
	it('should render phone/password auth', async () => {
		const { container, component } = render(AuthPassword, {
			props: { type: 'phone' }
		})
		expect(container).toMatchSnapshot()
		component.$on('click', handle)

		const phoneInput = container.querySelector('input[type="tel"]')
		const passwordInput = container.querySelector('input[type="password"]')
		const button = container.querySelector('button')

		await fireEvent.input(phoneInput, { target: { value: '+1 555-555-555' } })
		await fireEvent.input(passwordInput, { target: { value: 'password' } })
		await fireEvent.click(button)
		await tick()
		expect(handle).toHaveBeenDispatchedWith({
			phone: '+1 555-555-555',
			password: ''
		})
	})
})
