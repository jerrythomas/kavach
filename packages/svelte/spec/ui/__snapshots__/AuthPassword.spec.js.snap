// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AuthPassword.svelte > should render email/password auth 1`] = `
<div>
  <input-field
    class="flex flex-col input-email empty"
  >
    <label
      for="email"
    >
      email
    </label>
     
    <field
      aria-label="email"
      class="w-full flex flex-row items-center"
    >
      <icon
        aria-label="i-auth-email"
        class="flex flex-shrink-0 items-center justify-center "
        role="img"
        tabindex="-1"
      >
        <i
          aria-hidden="true"
          class="i-auth-email"
        />
      </icon>
       
       
      <input
        autocomplete="on"
        id="email"
        name="email"
        type="email"
      />
      
      
       
    </field>
     
  </input-field>
   
  <input-field
    class="flex flex-col input-password empty"
  >
    <label
      for="password"
    >
      Password
    </label>
     
    <field
      aria-label="Password"
      class="w-full flex flex-row items-center"
    >
      <icon
        aria-label="i-auth-password"
        class="flex flex-shrink-0 items-center justify-center "
        role="img"
        tabindex="-1"
      >
        <i
          aria-hidden="true"
          class="i-auth-password"
        />
      </icon>
       
       
      <input
        id="password"
        name="password"
        password=""
        type="password"
      />
      
      
       
    </field>
     
  </input-field>
   
  <button
    class="col-start-3 text-center h-10 mt-4 bg-primary text-white"
  >
    Sign in
  </button>
</div>
`;

exports[`AuthPassword.svelte > should render phone/password auth 1`] = `
<div>
  <input-field
    class="flex flex-col input-phone empty"
  >
    <label
      for="email"
    >
      phone
    </label>
     
    <field
      aria-label="phone"
      class="w-full flex flex-row items-center"
    >
      <icon
        aria-label="i-auth-phone"
        class="flex flex-shrink-0 items-center justify-center "
        role="img"
        tabindex="-1"
      >
        <i
          aria-hidden="true"
          class="i-auth-phone"
        />
      </icon>
       
       
      <input
        autocomplete="on"
        id="email"
        name="email"
        type="tel"
      />
      
      
       
    </field>
     
  </input-field>
   
  <input-field
    class="flex flex-col input-password empty"
  >
    <label
      for="password"
    >
      Password
    </label>
     
    <field
      aria-label="Password"
      class="w-full flex flex-row items-center"
    >
      <icon
        aria-label="i-auth-password"
        class="flex flex-shrink-0 items-center justify-center "
        role="img"
        tabindex="-1"
      >
        <i
          aria-hidden="true"
          class="i-auth-password"
        />
      </icon>
       
       
      <input
        id="password"
        name="password"
        password=""
        type="password"
      />
      
      
       
    </field>
     
  </input-field>
   
  <button
    class="col-start-3 text-center h-10 mt-4 bg-primary text-white"
  >
    Sign in
  </button>
</div>
`;
